-NO_OPENGL=1 breaks build configuration phase on macOS

I am trying to build a minimal, CPU-only version of OpenSubdiv. I'm on macOS 10.15.7. CMake 3.20.1.
```
cmake -DNO_EXAMPLES=1 -DNO_TUTORIALS=1 -DNO_REGRESSION=1 -DNO_PTEX=1 -DNO_DOC=1 -DNO_OPENCL=1 -DNO_OPENGL=1 -DNO_CLEW=1 -DNO_TBB=1 -DNO_METAL=1 -DNO_CUDA=1 ..
```
Is this expected?
```
<----snip---->
-- Configuring done
CMake Error at opensubdiv/CMakeLists.txt:320 (add_library):
  Error evaluating generator expression:

    $<TARGET_OBJECTS:osd_gpu_obj>

  Objects of target "osd_gpu_obj" referenced but no such target exists.


CMake Error at opensubdiv/CMakeLists.txt:273 (add_library):
  Error evaluating generator expression:

    $<TARGET_OBJECTS:osd_gpu_obj>

  Objects of target "osd_gpu_obj" referenced but no such target exists.


CMake Error at opensubdiv/CMakeLists.txt:320 (add_library):
  Error evaluating generator expression:

    $<TARGET_OBJECTS:osd_gpu_obj>

  Objects of target "osd_gpu_obj" referenced but no such target exists.


CMake Error at opensubdiv/CMakeLists.txt:273 (add_library):
  Error evaluating generator expression:

    $<TARGET_OBJECTS:osd_gpu_obj>

  Objects of target "osd_gpu_obj" referenced but no such target exists.


CMake Error at opensubdiv/CMakeLists.txt:320 (add_library):
  No SOURCES given to target: osd_dynamic_framework


CMake Error at opensubdiv/CMakeLists.txt:273 (add_library):
  No SOURCES given to target: osd_static_framework


CMake Generate step failed.  Build files cannot be regenerated correctly.
```

--- a/opensubdiv/CMakeLists.txt
+++ b/opensubdiv/CMakeLists.txt
@@ -312,7 +312,7 @@ if (NOT NO_LIB)
             $<TARGET_OBJECTS:far_obj>
             $<TARGET_OBJECTS:bfr_obj>
             $<TARGET_OBJECTS:osd_cpu_obj>
-            $<TARGET_OBJECTS:osd_gpu_obj>
+            $<$<TARGET_EXISTS:osd_gpu_obj>:$<TARGET_OBJECTS:osd_gpu_obj>>
             ${OPENGL_LOADER_OBJS}
         )
 
@@ -361,7 +361,7 @@ if (NOT NO_LIB)
                 $<TARGET_OBJECTS:far_obj>
                 $<TARGET_OBJECTS:bfr_obj>
                 $<TARGET_OBJECTS:osd_cpu_obj>
-                $<TARGET_OBJECTS:osd_gpu_obj>
+                $<$<TARGET_EXISTS:osd_gpu_obj>:$<TARGET_OBJECTS:osd_gpu_obj>>
                 ${OPENGL_LOADER_OBJS}
             )
 

